src/components/lab/canvas/CanvasMoleculeArea.js

```diff
@@ -0,0 +1,234 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Circle, Group } from 'react-konva';
+import {
+  CANVAS_MOLECULE_AREA_Y_POSITION,
+  CANVAS_MOLECULE_AREA_DEFAULT_RADIUS,
+  CANVAS_MOLECULE_AREA_DEFAULT_DASH,
+  CANVAS_MOLECULE_AREA_STROKE,
+  CANVAS_MOLECULE_AREA_EMPTY_FILL,
+  CANVAS_MOLECULE_AREA_STROKE_WIDTH,
+  CANVAS_MOLECULE_AREA_ACTIVE_DASH,
+  CANVAS_MOLECULE_AREA_EMPTY,
+  CANVAS_MOLECULE_AREA_ACTIVE,
+  CANVAS_MOLECULE_AREA_FULL,
+  CANVAS_MOLECULE_AREA_AWAITING_DELETE,
+  CANVAS_MOLECULE_AREA_ACTIVE_FILL,
+  CANVAS_MOLECULE_AREA_AWAITING_DELETE_FILL,
+  CANVAS_MOLECULE_AREA_AWAITING_DELETE_DASH,
+} from '../../../config/constants';
+import {
+  selectMoleculeInSideMenu,
+  displayMolecule,
+  changeMoleculeAreaStatus,
+  setIsPaused,
+} from '../../../actions';
+import CanvasMoleculeAreClearButton from './CanvasMoleculeAreaClearButton';
+
+const CanvasMoleculeArea = ({
+  x,
+  moleculeStatus,
+  containerIndex,
+  children,
+}) => {
+  const selectedMoleculeInSideMenu = useSelector(
+    ({ lab }) => lab.selectedMoleculeInSideMenu,
+  );
+  const moleculesOnCanvas = useSelector(({ lab }) => lab.moleculesOnCanvas);
+  const dispatch = useDispatch();
+
+  const onMouseEnter = (event) => {
+    const container = event.target.getStage().container();
+    container.style.cursor = 'pointer';
+  };
+
+  const onMouseLeave = (event) => {
+    const container = event.target.getStage().container();
+    container.style.cursor = 'default';
+  };
+
+  // main function/click handler for this component
+  const handleClick = () => {
+    // if some molecule has an active deletion area, clear that area
```

Comment:
I think it is not `clear that area` you wanted to write here^^

Reply:
Oh I see I was comparing with the next case, where you also mentioned `clear that area`. But since these two `clear that area` does not exactly mean the same, maybe add more details.

Reply:
In fact what you implemented is close to a state machine, and I'm wondering if there is a cleaner way to achieve this with react. Maybe we can discuss together.